facilitator/src/aggregation.rs

```diff
@@ -289,44 +289,64 @@ impl<'a> BatchAggregator<'a> {
         let mut processed_ingestion_packets = HashSet::new();
         let mut ingestion_packet_reader = ingestion_batch.packet_file_reader(&ingestion_header)?;
 
-        // Borrowing distinct parts of a struct works, but not under closures:
-        // https://github.com/rust-lang/rust/issues/53488
-        // The workaround is to borrow or copy fields outside the closure.
-        let logger = &self.logger;
-
+        // Convert ingestion packets into validation packets; if we can't
+        // decrypt a packet, drop/skip the entire ingestion batch without
+        // error. (Other errors are reported to the caller as normal & will
+        // cause the entire aggregation batch to fail.)
+        let mut ingestion_and_own_validation_packets = Vec::new();
         loop {
             let ingestion_packet =
                 match IngestionDataSharePacket::read(&mut ingestion_packet_reader) {
                     Ok(p) => p,
                     Err(Error::EofError) => break,
                     Err(e) => return Err(e.into()),
                 };
+            match ingestion_packet.generate_validation_packet(servers) {
+                Ok(p) => ingestion_and_own_validation_packets.push((ingestion_packet, p)),
+                Err(e) => {
+                    if let Some(Error::PacketDecryptionError(packet_uuid)) = e.downcast_ref() {
+                        // This ingestion batch contains a packet that can't be
+                        // decrypted. Ignore the entire ingestion batch, on the
+                        // assumption that our peer will do the same, since we
+                        // also wouldn't have been able to produce a peer
+                        // validation batch from this ingestion batch.
+                        warn!(self.logger, "Skipping ingestion batch due to undecryptable packet";
+                            event::PACKET_UUID => packet_uuid.to_string());
```

Comment:
Besides the log, we could emit a Prometheus metric for undecryptable packets, and maybe put an alert on it someday. I can live without it, though. We'll at least be able to search logs to tell that this has happened.

Reply:
For now, I'll leave it be -- there are a few other conditions that are warning-only right now.